backend/infrahub/git/repository.py

```diff
@@ -170,7 +171,10 @@ async def merge(self, source_branch: str, dest_branch: str, push_remote: bool =
         commit = self.get_commit_value(branch_name=source_branch, remote=False)
 
         try:
-            repo.git.merge(commit)
+            if config.SETTINGS.git.use_explicit_merge_commit:
+                repo.git.merge(commit, "--no-ff", m="Merged by Infrahub by Infrahub")
```

Comment:
Sorry I don't know if I duplicated this in my previous comment but the extra "by Infrahub" is a bit redundant here. We should start with just "Merged by Infrahub"